src/python/WMCore/Services/WorkQueue/WorkQueue.py

```diff
@@ -287,7 +287,8 @@ def updateElementsByWorkflow(self, workload, updateParams, status=None):
             # Update all workload parameters based on the full reqArgs dictionary
             workload.updateWorkloadArgs(updateParams)
             # Commit the changes of the current workload object to the database:
-            workload.saveCouchUrl(workload.specUrl())
+            metadata = {'name': wfName}
+            workload.saveCouch(self.hostWithAuth, self.db.name, metadata=metadata)
```

Comment:
Isn't this method used by global workqueue as well? It might be that making this change for the agent will actually break global workqueue.
In addition, I think it is the first time I see this update being done with metadata (and isn't it redundant with the `workload.updateWorkloadArgs`?)

Reply:
Yes (is used by global) and No (won't break anything) and No (it is not redundant)

Reply:
Right, I see this pattern has already been used in the method that updates the priority:
https://github.com/dmwm/WMCore/blob/180e27dde25189abd38f9035d7a50b442b6b8dbc/src/python/WMCore/Services/WorkQueue/WorkQueue.py#L253-L256